pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -39,6 +42,8 @@ def __init__(
         validate_data: bool = True,
         control_columns: Optional[List[str]] = None,
         yearly_seasonality: Optional[int] = None,
+        lagging_function: str = "geometric",
+        saturation_function: str = "logistic",
```

Comment:
Think that we should allow for class instances as well that fit our component definition and deprecate the `adstock_max_lag` as that is already part of the class. 

For instance, 
```python 
adstock= GeometricAdstock(max_lag=12)
mmm = DelayedSaturatedMMM(
    ...,
    adstock=adstock, 
    ..., 
)

Reply:
Considering considering!